how-it-works/1_protocol/04_execution.card.md

```diff
@@ -6,10 +6,15 @@ title: Execution Layer
 
 # Execution layer
 
-The execution layer processes one input at a time. This input is taken from one of the
-input queues, and is directed to one canister. Based on this input and the state of the
-canister, the execution environment updates the state of the canister, and additionally may
-add messages to output queues and update the ingress history (possibly with a response to
-an earlier ingress message).
+The execution layer, the topmost layer of the core IC protocol stack, is responsible for executing *queries* (the easy part) and scheduling and executing *canister messages* (the hard part).
+The execution layer implements a *Web Assembly (Wasm) virtual machine* for the execution of Wasm-based canister smart contract bytecode used on the IC.
```

Comment:
What is Wasm / web assembly? I mean, I know what it is, but maybe not all readers do...can we provide hyperlinks?